geomstats/geometry/discrete_surfaces.py

```diff
@@ -0,0 +1,687 @@
+        return gs.sqrt(gs.linalg.det(g))
+
+    def surface_metric_matrices(self, point):
+        """Compute the surface metric matrices.
+
+        The matrices are evaluated at the faces of a triangulated surface.
+
+        The surface metric is the pullback metric of the immersion q
+        defining the surface, i.e. of
+        the map q: M -> R3, where M is the parameterization manifold.
+
+        Parameters
+        ----------
+        point : array like, shape=[n_verticesx3]
+            One surface, i.e. the vertices of its triangulation.
+
+        Returns
+        -------
+        _ : array-like, shape=[n_faces, 2, 2]
+            Surface metric matrices evaluated at each face of
+            the triangulated surface.
+        """
+        one_forms = self.surface_one_forms(point)
+        transposed_one_forms = gs.transpose(one_forms, axes=(0, 2, 1))
+        return gs.matmul(one_forms, transposed_one_forms)
+
+
+class ElasticMetric(RiemannianMetric):
+    """Elastic metric defined a family of second order Sobolev metrics.
+
+    Each individual surface is represented by a 2d-array of shape `[
+    n_vertices, 3]`.
+
+    See [HSKCB2022]_ for details.
+
+    Parameters
+    ----------
+    space : Manifold
+        Instantiated DiscreteSurfaces manifold.
+    a0 : float
+        First order parameter.
+    a1 : float
+        Stretching parameter.
+    b1 : float
+        Shearing parameter.
+    c1 : float
+        Bending parameter.
+    d1 : float
+        additonal first order parameter.
+    a2 : float
+        Second order parameter.
+
+    References
+    ----------
+    .. [HSKCB2022] "Elastic shape analysis of surfaces with second-order
+    Sobolev metrics: a comprehensive numerical framework",
+    arXiv:2204.04238 [cs.CV], 25 Sep 2022
+    """
+
+    def __init__(self, space, a0, a1, b1, c1, d1, a2):
+        """Create a metric object."""
+        self.a0 = a0
+        self.a1 = a1
+        self.b1 = b1
+        self.c1 = c1
+        self.d1 = d1
+        self.a2 = a2
+        self.space = space
+        self.time_steps = 5
+
+    def inner_product(self, tangent_vec_a, tangent_vec_b, base_point):
+        """Inner product between two tangent vectors at a base point.
+
+        Parameters
+        ----------
+        tangent_vec_a: array-like, shape=[n_vertices, 3]
+            Tangent vector at base point.
+        tangent_vec_b: array-like, shape=[n_vertices, dim]
+            Tangent vector at base point.
+        base_point: array-like, shape=[n_vertices, dim]
+            Base point.
+
+        Returns
+        -------
+        inner_product : float
+            Inner-product.
+        """
+        h = tangent_vec_a
+        k = tangent_vec_b
+        point1 = base_point + h
+        point2 = base_point + k
+        norm = 0
+        if self.a0 > 0 or self.a2 > 0:
+            v_areas = self.space.vertex_areas(base_point)
+            if self.a2 > 0:
+                Del = self.space.get_laplacian(base_point)
```

Comment:
no uppercase letters in python + rename for readability
Del --> laplacian_at_base_point